### PR TITLE
Rename models (#101); improve type casting; add unit tests; update requirements.txt

### PR DESCRIPTION
**Pull request by acrellin**
E.g. `RFC` becomes `Random Forest Classifier`

### CODE REVIEW

File: mltsp/tests/test_predict.py [hunk 1]
```diff
@@ -58,8 +58,8 @@ def test_model_predictions():
     """Test inner model prediction function"""
     featureset = xray.open_dataset(pjoin(cfg.FEATURES_FOLDER,
                                          'test_featureset.nc'))
-    model = build_model.build_model_from_featureset(featureset,
-                                                    model_type='RFC')
+    model = build_model.build_model_from_featureset(
+        featureset,model_type='Random Forest Classifier')
```
Comment (bnaul) [inline mltsp/tests/test_predict.py]: Missing a space after the comma.

**Comment by bnaul** [comment]
:+1:

**Comment by bnaul** [comment]
Does `util.cast_model_params` still work with these new names? It looks like it references `entry["abbr"]` from `model_descriptions` which I think no longer exists.

**Comment by acrellin** [comment]
@bnaul good catch, this wasn't getting caught b/c the frontend tests aren't being run.... Reminds me to add a test for `util.cast_model_params`

**Comment by bnaul** [comment]
Yeah was just gonna suggest that, it's a complicated function so it would be good to test all the various permutations.

**Comment by acrellin** [comment]
The built-in `ast.literal_eval` is safe and simplifies the type-casting function.

**Comment by acrellin** [comment]
I'm glad I looked more closely into the type-casting function and added unit tests; dictionaries were not being parsed and cast correctly before!

**Comment by bnaul** [comment]
Sounds like a good approach to me; glad this is properly tested now!

**Comment by acrellin** [comment]
Looks like merging that last PR create conflicts... I'll take care of that, and @stefanv will you look this over when you get a chance?

**Comment by bnaul** [comment]
Oops...is it just requirements.txt? I tried to not overlap at all but I
guess I missed something. But yeah, let's get these in.

**Comment by acrellin** [comment]
Should be good to go now.

File: mltsp/util.py [hunk 2]
```diff
@@ -108,30 +126,27 @@ def cast_model_params(model_type, model_params):
             if p["name"] == k:
                 param_entry = p
                 break
-        # If type description is a single type and not of type list, do cast
-        if type(param_entry["type"]) == type and param_entry["type"] != list:
+        # If type description is a single type and not of type str, do cast
+        if type(param_entry["type"]) == type and param_entry["type"] != str:
```
Comment (bnaul) [inline mltsp/util.py]: `isinstance(param_entry["type"], type)`?

File: mltsp/util.py [hunk 3]
```diff
@@ -108,30 +126,27 @@ def cast_model_params(model_type, model_params):
...
-        elif param_entry["type"] == list:
-            model_params[k] = model_params[k].replace("[", "").replace("]", "")\
-                                                              .replace(" ", "")\
-                                                              .split(",")
-            for i in range(len(model_params[k])):
-                if "." in model_params[k][i]:
-                    model_params[k][i] = float(model_params[k][i])
-                elif model_params[k][i].isdigit():
-                    model_params[k][i] = int(model_params[k][i])
+            if isinstance(ast.literal_eval(model_params[k]), dest_type):
```
Comment (bnaul) [inline mltsp/util.py]: Catch exception for bad input.

**Comment by acrellin** [comment]
Alright, folks - everyone happy? Please merge before you push up more conflict-making changes! :wink:

**Comment by bnaul** [comment]
:+1:

**Comment by bnaul** [comment]
Looks like `.drone/build.sh` got missed with due to our master shenanigans; if you rebase this branch it should be fine I think.

**Pull request merged**